tests/wpt_tests.cpp

```diff
@@ -495,70 +495,54 @@ bool urltestdata_encoding(const char *source) {
 
-          std::string_view host;
-          if (!object["host"].get_string().get(host)) {
-            TEST_ASSERT(input_url->get_host(), host,
-                        "Hostname " + element_string + input_url->to_string());
-          }
+          std::string_view host = object["host"].get_string().value();
+          TEST_ASSERT(input_url->get_host(), host,
+                      "Host " + element_string + input_url->to_string());
 
-          std::string_view hostname;
-          if (!object["hostname"].get_string().get(hostname)) {
-            TEST_ASSERT(input_url->get_hostname(), hostname,
-                        "Hostname " + element_string + input_url->to_string());
-          }
+          std::string_view hostname = object["hostname"].get_string().value();
+          TEST_ASSERT(input_url->get_hostname(), hostname,
+                      "Hostname " + element_string + input_url->to_string());
 
-          std::string_view port;
-          if (!object["port"].get_string().get(port)) {
-            TEST_ASSERT(input_url->get_port(), port, "Port " + element_string);
-          }
+          std::string_view port = object["port"].get_string().value();
+          TEST_ASSERT(input_url->get_port(), port, "Port " + element_string);
 
-          std::string_view pathname{};
-          if (!object["pathname"].get_string().get(pathname)) {
-            TEST_ASSERT(input_url->get_pathname(), pathname,
-                        "Pathname " + element_string + input_url->to_string());
-          }
-          std::string_view query;
-          if (!object["query"].get(query)) {
-            TEST_ASSERT(input_url->get_search(), query,
-                        "Query " + element_string + input_url->to_string());
-          }
+          std::string_view pathname = object["pathname"].get_string().value();
+          TEST_ASSERT(input_url->get_pathname(), pathname,
+                      "Pathname " + element_string + input_url->to_string());
 
-          std::string_view hash;
-          if (!object["hash"].get_string().get(hash)) {
-            TEST_ASSERT(
-                input_url->get_hash(), hash,
-                "Hash/Fragment " + element_string + input_url->to_string());
-          }
+          std::string_view search = object["search"].get_string().value();
+          TEST_ASSERT(input_url->get_search(), search,
+                      "Search " + element_string + input_url->to_string());
 
-          std::string_view href;
-          if (!object["href"].get_string().get(href)) {
-            TEST_ASSERT(input_url->get_href(), href,
-                        "href " + element_string + input_url->to_string());
-          }
+          std::string_view hash = object["hash"].get_string().value();
+          TEST_ASSERT(
+              input_url->get_hash(), hash,
+              "Hash/Fragment " + element_string + input_url->to_string());
+
+          std::string_view href = object["href"].get_string().value();
```

Comment:
I don't want you to change this because it is too small to matter, but you can just do...

```C++
std::string_view href = object["href"].get_string();
```

And it throws if the cast is not possible.

Reply:
👍 Better I do it now, than later